dsp-shared/src/main/scala/dsp/valueobjects/LangString.scala

```diff
@@ -53,13 +55,104 @@ object LangString {
+   * Add a set of LangStrings to the MultiLangString
+   *
+   * @param values the new LangStrings
+   * @return either a MultiLangString or a ValidationException
+   */
+  def addLangStrings(values: Set[LangString]): Validation[ValidationException, MultiLangString] =
+    MultiLangString.make(self.langStrings ++ values)
+
+  def removeLanguage(value: LanguageCode): Validation[ValidationException, MultiLangString] =
+    // TODO-BL: [discuss] what should happen if the element is not contained?
+    MultiLangString.make(self.langStrings.filter(langString => langString.language == value))
+
+  def removeLangString(value: LangString): Validation[ValidationException, MultiLangString] =
+    // TODO-BL: [discuss] what should happen if the element is not contained?
+    MultiLangString.make(self.langStrings - value)
+
+  def removeLangStrings(values: Set[LangString]): Validation[ValidationException, MultiLangString] =
+    // TODO-BL: [discuss] what should happen if the elements are not contained?
+    MultiLangString.make(self.langStrings -- values)
+
+  /**
+   * Updates the LangString of a certain language to a new value.
+   *
+   * @param value the LangString containing the new value
+   * @return the updated MultiLangString
+   */
+  def updateLangString(value: LangString): Validation[ValidationException, MultiLangString] =
+    // TODO-BL: [discuss] what should happen if the element is not contained?
+    MultiLangString.make(self.langStrings.filter(langString => langString.language != value.language) + value)
+
+  /**
+   * Returns the LangString with the specified LanguageCode, if available.
+   *
+   * @param lang the requested language
+   * @return Optionally the LangString in the requested language
+   */
+  def getByLanguage(lang: LanguageCode): Option[LangString] =
+    self.lookup.get(lang)
+}
+
+object MultiLangString {
+
+  /**
+   * Creates a [[zio.prelude.Validation]] that either fails with a ValidationException, or succeeds with a MultiLangString.
+   * Ensures that the MultiLangString is not empty and that the languages are unique.
+   *
+   * @param values a set of LangString value objects
+   * @return a validation of a MultiLangString value object
+   */
+  def make(values: Set[LangString]): Validation[ValidationException, MultiLangString] =
+    values match {
+      case v if v.isEmpty => Validation.fail(ValidationException(MultiLangStringErrorMessages.MultiLangStringEmptySet))
+      case v if v.size > v.map(_.language).size =>
+        val languages = v.toList.map(_.language)
+        val languageCount = languages.foldLeft[Map[LanguageCode, Int]](Map.empty) { (acc, lang) =>
+          acc.updated(lang, acc.getOrElse(lang, 0) + 1)
+        }
+        val nonUniqueLanguages = languageCount.filter { case (_, count) => count > 1 }.keySet
+        Validation.fail(ValidationException(MultiLangStringErrorMessages.LanguageNotUnique(nonUniqueLanguages)))
+      case _ => Validation.succeed(new MultiLangString(values) {})
+    }
+}
```

Comment:
I'm not quite sure what is going on here. Can LangString exist w/o String or ISO? Maybe it would be good to add docs to all methods.

Reply:
removed the entire mess. Now it should be clearer :)